api/policies/UserPolicy.js

```diff
@@ -92,7 +55,32 @@ module.exports = {
     throw Boom.unauthorized('You are not allowed to do this operation');
   },
 
-  canUpdate,
+  async canUpdate(request) {
+    // If the acting user is an admin
+    //   OR any user is targeting themselves
+    // THEN are allowed to update the target user.
+    if (request.auth.credentials.is_admin || request.auth.credentials.id === request.params.id) {
+      const user = await User.findById(request.params.id);
+      if (!user) {
+        logger.warn(
+          `[UserPolicy->canUpdate] User ${request.params.id} not found`,
+        );
+        throw Boom.notFound();
+      }
```

Comment:
This breaks the [`/api/v3/user` endpoint (GET)](https://github.com/UN-OCHA/hid_api/blob/9fd38d6ca44d974c17bab359d525110cb6556ffd/config/routes.js#L376): we get a 404 (because no user id) instead of the full list of users when accessed with an admin user (we get a 403 as expected for a non admin user).

Maybe we can still use the `canFind` user policy for this endpoint and if there is an `id` provided return the result of `canUpdate`, or explicitly separate the `/api/v3/user` and `/api/v3/user/{id}` endpoints and use `canFind` for the former and `canUpdate` for the latter. What do you think?

Reply:
Thanks, can't believe I forgot to re-test that. Yes, I will simply split the route definition into two, so we can keep the old policy when no ID is submitted.